cmd/logger.go

```diff
@@ -137,7 +108,32 @@ func getSource() string {
 	return fmt.Sprintf("[%s:%d:%s()]", filename, lineNum, funcName)
 }
 
-func logIf(level logrus.Level, source string, err error, msg string, data ...interface{}) {
+// trace method - to create stack trace
+func getTrace(level int) []string {
+	var trace []string
+	var pc uintptr
+	var file string
+	var lineNumber int
+	ok := true
+	pc, file, lineNumber, ok = runtime.Caller(level)
+	rootPath := filepath.Dir(file) + string(os.PathSeparator)
+	for ok {
+		// Clean up the common root path, rootPath,
+		// from the collected stack trace
+		funcName := runtime.FuncForPC(pc).Name()
+		_, funcName = filepath.Split(funcName)
+		file = strings.TrimPrefix(file, rootPath)
+		elem := fmt.Sprintf("%v:%v:%v", file, lineNumber, funcName)
```

Comment:
does `funcName` come with `()` suffixed? else we would need to add `()`

Reply:
`funcName` comes without `()` suffixed.
We can append `()` to the end of `funcName`, that is not a problem, but wouldn't it be confusing if we print out function names with only `()` while the function might have some arguments which is supposed to appear inside parenthesis ?
Maybe we can append `(...)`, instead of `()` simply. But even one can argue that it would be confusing for functions that get no arguments. What do you think?